[apache] Add collection for JBoss Core Services and Red Hat Software …

msfm opened the pull request:
…Collections

Signed-off-by: Masafumi Miura <mmiura@redhat.com>

pmoravec commented on sos/plugins/apache.py:
```diff
@@ -51,6 +51,14 @@ def setup(self):
+            # Red Hat Software Collections
+            "/opt/rh/httpd24/root/etc/httpd/conf/httpd.conf",
+            "/opt/rh/httpd24/root/etc/httpd/conf.d/*.conf",
+            "/opt/rh/httpd24/root/etc/httpd/conf.modules.d/*.conf",
```
Can't either added config file contain some secret that we should obfuscate?

msfm replied:
These config files are similar to other's httpd config files. (It's just located under different diretory "/opt/rh" in order to avoid conflict to default one.) So, I think these files do not contain secret information that we should obfuscate.

pmoravec commented on sos/plugins/apache.py:
```diff
@@ -65,18 +73,25 @@ def setup(self):
-                "/var/log/httpd24/*"
+                "/var/log/httpd24/*",
+                # Red Hat JBoss Core Services
+                "/opt/rh/jbcs-httpd24/root/var/log/httpd/*"
```
Couldn't this directory contain many big logfiles, causing sosreport to collect too much data? (anyway since this plugin collects whole `/var/log/httpd2[2,4]/` directories, I would expect the answer is no)

msfm replied:
log files under this directory should be maintained by logrotate but it appears that JBCS is missing logrotate setting for this directory. (This is an issue in JBCS). I will just remove this from this PR until logrotate setting is fixed in JBCS.

pmoravec commented:
See the pep8 failures in Travis CI build:

    sos/plugins/apache.py:82:80: E501 line too long (89 > 79 characters)
    sos/plugins/apache.py:83:80: E501 line too long (88 > 79 characters)
    sos/plugins/apache.py:84:80: E501 line too long (93 > 79 characters)
    sos/plugins/apache.py:85:80: E501 line too long (92 > 79 characters)

(I suggest running "pep8 sos" before comitting, or even better to have a commit hook doing the same automatically)

pmoravec commented:
Also it makes sense to shrink the commit subject to "[apache] Add collection for JBoss Core Services and RHCS" to fit subject length limit.

No new PR is required, just modify your current commit accordingly and forcefully push to the same private branch.

bmr-cymru commented:
This really needs some consideration as to whether it is appropriate to just jam it into the Apache plugin. I'm not convinced that it is right now, but I'll take a more detailed look this week.

msfm commented:
I pushed with the following updates:

- Removed `/opt/rh/jbcs-httpd24/root/var/log/httpd/*`
- Checked pep8 and fixed `"E501 line too long"`
- Modfied the commit subject to use abbreviated name "JBCS (JBoss Core Services)" and "RHSCL (Red Hat Software Collections)"

msfm commented:
> This really needs some consideration as to whether it is appropriate to just jam it into the Apache plugin. I'm not convinced that it is right now, but I'll take a more detailed look this week.

A customer does not always understand what Apache httpd is running and often just recognizes Apache httpd provided by Red Hat. IMHO, it's better to capture all possible information at once and it can help to avoid asking for collecting information many times.

bmr-cymru commented:
> A customer does not always understand what Apache httpd is running and often just recognizes Apache
> httpd provided by Red Hat. IMHO, it's better to capture all possible information at once and it can help to
> avoid asking for collecting information many times.

That is why we have automatic plugin triggers: the user is not expected to know or care where the data comes from. That's our responsibility.

You should look at the proposed support for native SCL plugins - that seems like a more appropriate location for these additions (pull request #900).

bmr-cymru commented:
Needs conversion to SCLPlugin.

bmr-cymru commented:
No update: moving out to 3.7.

TurboTurtle commented:
It has been nearly 3 years since the last activity on this PR, so I am closing it at this point.

If these changes are still needed, please rebase against the current master and either open a new PR or re-open this one.
